File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        // for last element of nums1
        int midx = m - 1;
        // for last element of nums2
        int nidx = n - 1;
        // for index of reault array
        int right = m + n - 1;
        // to comapare two arrays and sort them
        while(nidx >= 0){
            if( midx >= 0 && nums1[midx] > nums2[nidx] ){
                nums1[right] = nums1[midx];
                midx--;
            }
            else{
                nums1[right] = nums2[nidx];
                nidx--;
            }
            right--;
        }
    }
}

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        midx = m - 1
        nidx = n - 1
        idx = m + n - 1

        while midx >= 0 and nidx >= 0:
            if (nums1[midx] > nums2[nidx]):
                nums1[idx] = nums1[midx]
                midx -= 1
            else:
                nums1[idx] = nums2[nidx]
                nidx -= 1
            idx -= 1
        
        while nidx >= 0:
            nums1[idx] = nums2[nidx]
            nidx -= 1
            idx -= 1